src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -191,6 +212,12 @@ void IServiceBehavior.ApplyDispatchBehavior(ServiceDescription description, Serv
                         {
                             ApplyAuthorizationPoliciesAndManager(behavior);
                         }
+
+                        if (_isAuthorizationServiceSet)
+                        {
+                            behavior.AuthorizationService = _authorizationService;
+                            behavior.RequireClaimsPrincipalOnOperationContext = true;
+                        }
```

Comment:
Doesn't _isAuthorizationServiceSet get set to true if there's an IAuthorizationPolicyProvider available in DI? If they have it available but aren't using it for their CoreWCF service, won't this will cause an exception because the ClaimsPrincipal isn't available?

Reply:
Indeed. In case user added Authorization services in DI but does not use it fore CoreWCF we will have `RequireClaimsPrincipalOnOperationContext` set to true which is not correct, I will add a unit test to highlight it.
I still need to find a way to set `RequireClaimsPrincipalOnOperationContext` to true when `HttpClientCredentialType.InheritedFromHost` is configured in the binding.

Reply:
fixed in https://github.com/CoreWCF/CoreWCF/pull/850/commits/8da1ca6e9f916bf3eb6a64f0fccdd56c859fc84d